sys/net/network_layer/ng_ipv6/ng_ipv6.c

```diff
@@ -432,7 +442,13 @@ static void _send(ng_pktsnip_t *pkt, bool prep_hdr)
     else {
         ng_ipv6_addr_t *next_hop = NULL;
 
-        next_hop = &hdr->dst;   /* TODO: next hop determination */
+#ifdef MODULE_NG_IPV6_EXT_RH
+        next_hop = ng_ipv6_ext_rh_next_hop(hdr);
```

Comment:
> is it really necessary to iterate over all next headers in both functions ng_ipv6_ext_demux and ng_ipv6_ext_rh_next_hop?

`ng_ipv6_ext_demux()` is only called on receive if the packet is for the node, `ng_ipv6_ext_rh_next_hop()` is called on send. Those are two pretty distinct cases.